client/src/graphics/window.rs

```diff
@@ -256,6 +256,16 @@ impl Window {
                                 sim.toggle_no_clip();
                             }
                         }
+                        VirtualKeyCode::Key1 | VirtualKeyCode::Key2 | VirtualKeyCode::Key3
+                        | VirtualKeyCode::Key4 | VirtualKeyCode::Key5 | VirtualKeyCode::Key6
+                        | VirtualKeyCode::Key7 | VirtualKeyCode::Key8 | VirtualKeyCode::Key9
+                        | VirtualKeyCode::Key0 => {
+                            if state == ElementState::Pressed {
+                                if let Some(sim) = self.sim.as_mut() {
+                                    sim.select_material(number_key_to_index(key));
+                                }
+                            }
+                        }
```

Comment:
nit: We could avoid listing every key twice by moving this logic into the catch-all `_ => {}` case below, modifying `number_key_to_index` to return `Option`, and calling it there.